test/lit/passes/ldse.wast

```diff
@@ -0,0 +1,1481 @@
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT: )
+ (func $memory-wrong-const
+  (i32.store
+   (i32.const 10)
+   (i32.const 20)
+  )
+  (i32.store
+   (i32.const 30)
+   (i32.const 40)
+  )
+ )
+
+ ;; CHECK:      (func $memory-wrong-offset
+ ;; CHECK-NEXT:  (i32.store
+ ;; CHECK-NEXT:   (i32.const 10)
+ ;; CHECK-NEXT:   (i32.const 20)
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT:  (i32.store offset=1
+ ;; CHECK-NEXT:   (i32.const 10)
+ ;; CHECK-NEXT:   (i32.const 30)
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT: )
+ (func $memory-wrong-offset
+  (i32.store
+   (i32.const 10)
+   (i32.const 20)
+  )
+  (i32.store offset=1
+   (i32.const 10)
+   (i32.const 30)
+  )
+ )
+
+ ;; CHECK:      (func $memory-wrong-size
+ ;; CHECK-NEXT:  (i32.store
+ ;; CHECK-NEXT:   (i32.const 10)
+ ;; CHECK-NEXT:   (i32.const 20)
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT:  (i32.store16
+ ;; CHECK-NEXT:   (i32.const 10)
+ ;; CHECK-NEXT:   (i32.const 30)
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT: )
+ (func $memory-wrong-size
+  (i32.store
+   (i32.const 10)
+   (i32.const 20)
+  )
+  (i32.store16
+   (i32.const 10)
+   (i32.const 30)
+  )
+ )
+
+ ;; CHECK:      (func $memory-other-interference
+ ;; CHECK-NEXT:  (i32.store
+ ;; CHECK-NEXT:   (i32.const 10)
+ ;; CHECK-NEXT:   (i32.const 20)
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT:  (memory.fill
+ ;; CHECK-NEXT:   (i32.const 0)
+ ;; CHECK-NEXT:   (i32.const 0)
+ ;; CHECK-NEXT:   (i32.const 30)
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT:  (i32.store
+ ;; CHECK-NEXT:   (i32.const 10)
+ ;; CHECK-NEXT:   (i32.const 30)
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT: )
+ (func $memory-other-interference
+  (i32.store
+   (i32.const 10)
+   (i32.const 20)
+  )
+  (memory.fill
+   (i32.const 0)
+   (i32.const 0)
+   (i32.const 30)
+  )
+  (i32.store
+   (i32.const 10)
+   (i32.const 30)
+  )
+ )
+
+ ;; CHECK:      (func $memory-load
+ ;; CHECK-NEXT:  (i32.store
+ ;; CHECK-NEXT:   (i32.const 10)
+ ;; CHECK-NEXT:   (i32.const 20)
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT:  (drop
+ ;; CHECK-NEXT:   (i32.load
+ ;; CHECK-NEXT:    (i32.const 10)
+ ;; CHECK-NEXT:   )
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT:  (i32.store
+ ;; CHECK-NEXT:   (i32.const 10)
+ ;; CHECK-NEXT:   (i32.const 30)
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT: )
+ (func $memory-load
+  (i32.store
+   (i32.const 10)
+   (i32.const 20)
+  )
+  (drop
+   (i32.load
+    (i32.const 10)
+   )
+  )
+  (i32.store
+   (i32.const 10)
+   (i32.const 30)
+  )
+ )
+
+ ;; CHECK:      (func $memory-load-wrong-offset
+ ;; CHECK-NEXT:  (i32.store
+ ;; CHECK-NEXT:   (i32.const 10)
+ ;; CHECK-NEXT:   (i32.const 20)
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT:  (drop
+ ;; CHECK-NEXT:   (i32.load offset=1
+ ;; CHECK-NEXT:    (i32.const 10)
+ ;; CHECK-NEXT:   )
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT:  (i32.store
+ ;; CHECK-NEXT:   (i32.const 10)
+ ;; CHECK-NEXT:   (i32.const 30)
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT: )
+ (func $memory-load-wrong-offset
+  (i32.store
+   (i32.const 10)
+   (i32.const 20)
+  )
+  (drop
+   (i32.load offset=1
+    (i32.const 10)
+   )
+  )
```

Comment:
What does this try to test? If there's no `offset`, I think this is still not optimizable because this will load from the same address with the first store, no? Isn't this unoptimizable just because we consider all memory reading/writing operations, regardless of their addresses, as `mayIteract`s?